src/renderers/shared/ReactPerfDev.js

```diff
@@ -0,0 +1,413 @@
+/**
+ * Copyright 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactPerfDev
+ */
+
+'use strict';
+
+var ReactDebugToolDev = require('ReactDebugToolDev');
+var warning = require('warning');
+
+var alreadyWarned = false;
+
+var warnInProductionAndReturnDefault = (val) => () => {
+  if (alreadyWarned) {
+    return val;
+  }
+  alreadyWarned = true;
+  if (typeof console !== 'undefined') {
+    console.error(
+      'ReactPerf is not supported in the production builds of React. ' +
+      'To collect measurements, please use the development build of React instead.'
+    );
+  }
+  return val;
+};
+
+var ReactPerfDev = {
```

Comment:
@gaearon would you take a look at this? `ReactPerf` is the only dev-only module that we publicly expose and it was refactored to make sure the public API doesn't break.